src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/IRuleProvider.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+    /// <summary>
+    /// Enable binding rules (like <see cref="IBindingProvider"/> to be self-describing for tooling. 
+    /// </summary>
+    internal interface IRuleProvider
```

Comment:
Basically yes. But using the v2 binding interfaces (BindToInput, BindToCollecotr, BindToStream), this happens automatically because the extension author no longer implements IBindingProvider directly anymore.